src/Lexer.ts

```diff
@@ -0,0 +1,98 @@
+// @author Anush kumar N <anush.kumar@codingmart.com>
+
+import TokenType from './Token';
+
+type arrayOfString = Array<string>;
+type literalWithType = { type: Symbol; literal: string | undefined };
```

Comment:
Why allow undefined in the literal? The reference implementation in Swift managed to maintain full type system integrity here; surely the same can be achieved in Typescript?

Reply:
yes i will update the code